### PR TITLE
Feature ci kubernetes

### PR DESCRIPTION
**Pull request by CyberCuCuber**
**Helm:**

-   service
-   deployment
-   ingress
-   configmap
-   service account
-   service account role

**Jenkins:**

-   added deploy master branch in k8s cluster

_**link to health check:**_

-   _https://ivanoff.smodata.net/authn/api/v1/authn/health_check_

### CODE REVIEW

File: k8s/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          ports:
+            - containerPort: {{ .Values.deployment.webserver.containerPort }}
+              name: {{ .Values.deployment.webserver.portName }}
+              protocol: {{ .Values.deployment.webserver.protocol }}
+          volumeMounts:
+            - name: {{ .Values.deployment.webserver.volumeName }}
+              mountPath: {{ .Values.deployment.webserver.volumeMountPath }}
+        - name: {{ .Values.deployment.app.name }}
+          image: {{ .Values.deployment.app.image }}
+          env:
```
Comment (ggramal) [inline k8s/templates/deployment.yaml:40]: lets make envs settable because they can be different from env to env

File: k8s/values.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+general:
```
Comment (ggramal) [inline k8s/values.yaml:1]: there should not be only one value file but per env value file

File: k8s/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        - name: {{ .Values.deployment.webserver.name }}
+          image: {{ .Values.deployment.webserver.image }}
+          ports:
+            - containerPort: {{ .Values.deployment.webserver.containerPort }}
+              name: {{ .Values.deployment.webserver.portName }}
+              protocol: {{ .Values.deployment.webserver.protocol }}
+          volumeMounts:
+            - name: {{ .Values.deployment.webserver.volumeName }}
+              mountPath: {{ .Values.deployment.webserver.volumeMountPath }}
+        - name: {{ .Values.deployment.app.name }}
```
Comment (ggramal) [inline k8s/templates/deployment.yaml:38]: we also need to add liveness and readiness probes

File: k8s/templates/deployment.yaml (hunk 1 above)
Comment (CyberCuCuber) [reply k8s/templates/deployment.yaml:40]: @ggramal ok, done

File: k8s/values.yaml (hunk 2 above)
Comment (CyberCuCuber) [reply k8s/values.yaml]: @ggramal ok, done

File: k8s/templates/deployment.yaml (hunk 3 above)
Comment (CyberCuCuber) [reply k8s/templates/deployment.yaml:38]: @ggramal done

**Pull request merged**